my_sprites.py

```diff
@@ -76,14 +94,38 @@ def go_to_position(self, target_pos: tuple[int, int]):
 
     def on_update(self, delta_time: float = 1 / 60):
 
-        if not self.path:
+        # state control
+        if arcade.has_line_of_sight(self.position, self.target.position, self.barriers.blocking_sprites):
+            self.state = EnemyState.CHASING
+        else:
+            self.state = EnemyState.ROAMING
```

Comment:
When you have added a setter for _self.state_, like I asked about above, we can trigger emotes when the state changes here.

Reply:
sure